src/System.Runtime.Extensions/src/System/BitConverter.cs

```diff
@@ -364,20 +364,39 @@ public static string ToString(byte[] value, int startIndex, int length)
             }
 
             int chArrayLength = length * 3;
+            const int StackLimit = 512; // arbitrary limit to switch from stack to heap allocation
```

Comment:
Would catching stackoverflow allow for recovery if this had happened too deep (and perhaps allow for a larger size limit) or would the way that it's in some ways an unusual exception mean that wouldn't work?